File: Python/Program/FruitCat.py
from PetCat import PetCat
from Fruit import Fruit


class FruitCat(PetCat, Fruit):
    def __init__(self, gender, toy, name, fruit_name, fruit_color):
        PetCat.__init__(self, gender, toy, name)
        Fruit.__init__(self, fruit_name, fruit_color)
        self.is_crying = False

    def set_is_crying(self, crying):
        self.is_crying = crying

    def get_is_crying(self):
        return self.is_crying

    def display(self):
        print("FRUITCAT")
        self.display_info()

    def display_info(self):
        PetCat.display_info(self)
        print(f"> Crying: {'Yes' if self.is_crying else 'No'}")
        Fruit.display_info(self)

File: Java/Program/FruitCat.java

class FruitCat extends PetCat {
    private boolean isCrying;
    private Fruit fruit;

    public FruitCat(String gender, String toy, String name, String fruitName, String fruitColor) {
        super(gender, toy, name);
        this.fruit = new Fruit(fruitName, fruitColor) {
        };
        this.isCrying = false;
    }

    public void setIsCrying(boolean crying) {
        this.isCrying = crying;
    }

    public boolean getIsCrying() {
        return isCrying;
    }

    @Override
    public void display() {
        System.out.println("FRUITCAT");
        displayInfo();
    }

    @Override
    public void displayInfo() {
        super.displayInfo();
        System.out.println("> Crying: " + (isCrying ? "Yes" : "No"));
        fruit.displayInfo();
    }
}